src/Result.php

```diff
@@ -261,14 +262,23 @@ private function ensureHasKeyValue(): void
     /**
      * BC layer for a wide-spread use-case of old DBAL APIs
      *
-     * @deprecated This API is deprecated and will be removed after 2022
+     * @deprecated Use {@see fetchNumeric()}, {@see fetchAssociative()} or {@see fetchOne()} instead.
+     *
+     * @psalm-param FetchMode::* $mode
      *
      * @return mixed
      *
      * @throws Exception
      */
     public function fetch(int $mode = FetchMode::ASSOCIATIVE)
     {
+        Deprecation::trigger(
+            'doctrine/dbal',
+            'https://github.com/doctrine/dbal/pull/5948',
```

Comment:
For methods in this file, I think https://github.com/doctrine/dbal/pull/4007 should be used.